File: CPP/DivisibleSubArrays[CB].cpp
#include <bits/stdc++.h>
using namespace std;

#define ll long long

int main() {
    ll t;
    cin>>t;
    
    while(t--) {
        ll n;
        cin>>n;

        vector<ll> num(n), fre(n);
        ll sum{};
        
        fre[0] = 1;
        for(int i{}; i<n; i++) {
            cin>>num[i];
            sum +=num[i];
            sum %= n;
            sum = (sum + n) % n;
            fre[sum]++;
        }

        ll ans{};
        for(int i{}; i<n; i++) {
            ans += fre[i] * (fre[i] - 1) / 2;
        }

        cout<<ans<<endl;
    }
}

File: Python/DivisibleSubArrays[CB].py
t = int(input())

while(t) :
    n = int(input())
    num = []
    fre = [0] * n
    fre[0] = 1

    sum = 0
    num = list(map(int, input().split()))
    for i in range(n):
        sum += num[i]
        sum = (sum + n) % n
        fre[sum] += 1

    ans = 0
    for each in fre:
        ans += each * (each - 1) / 2
    print(int(ans))
    t -= 1
